java/memory/memory-netty/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -1091,6 +1099,32 @@ public void testMemoryLeakWithReservation() throws Exception {
     }
   }
 
+  @Test
+  public void testMemoryUsage() {
+    ListAppender<ILoggingEvent> memoryLogsAppender = new ListAppender<>();
+    Logger logger = (Logger) LoggerFactory.getLogger("arrow.allocator");
+    logger.setLevel(Level.TRACE);
+    logger.addAppender(memoryLogsAppender);
+    memoryLogsAppender.start();
+    try (ArrowBuf buf = new ArrowBuf(ReferenceManager.NO_OP, null,
+        1024, new PooledByteBufAllocatorL().empty.memoryAddress())) {
+      buf.memoryAddress();
```

Comment:
What I'm saying is, you have no guarantee the logger will actually have logged anything by the time the next few lines run. So this is just adding a flaky test.

Reply:
And IMO, the easier way to test this would be to factor out a function that generates the log string, test that function, and use that function from the thread, rather than hooking into the physical logger.

Reply:
Validation inside a loop